Rollbar/RollbarLogger.cs

```diff
@@ -510,6 +510,12 @@ void IDisposable.Dispose()
             SemaphoreSlim signal = null
             )
         {
+            // adding logic here that would prevent potential flooding with the tasks when reaching the application thread pool starvation for any reason: 
+            if (this._pendingTasks.Count > this.Config.ReportingQueueDepth)
```

Comment:
This looks good to me as it is solving the issue. However, this will drop the newest instead of the oldest which is what we would get from a fixed size queue. I think we should merge this but we can reconsider later what the behavior could me.

As an aside, are we ignoring duplicate errors? In most other SDKS, if I call `rollbar.error(someException); rollbar.error(someException);` we will deduplicate that and only send it once. How we do the deduplication depends on the library and language, but without it sometimes browser javascript would get into a loop at send 10k+ identical errors in a very small time window.

Reply:
Re that aside: in most other libraries the deduplication is only on the last call, i.e.
```
rollbar.error(someExc);
rollbar.error(differentExc);
rollbar.error(someExc);
```
results in 3 errors getting sent

Reply:
yes, once we move away from Tasks usage for packaging the fixed depth transmission queue will be the only defining factor (for what gets omitted and in what order).
At the moment there is no attempt to ignore duplicate payloads. Adding something like this would mean iterating transmission queue every time on incoming payload trying to find a dupe there before actually en-queuing the incoming one. That could easily become an expensive operation. In .NET, I can try if getting an exception hash code can help in matching same exceptions.